protocols/imap/src/main/java/org/apache/james/imap/processor/GetAnnotationProcessor.java

```diff
@@ -92,18 +92,28 @@ private void respond(String tag, ImapCommand command, Responder responder, Strin
         }
     }
 
-    private Optional<Integer> getMaxSizeValue(List<MailboxAnnotation> mailboxAnnotations, long maxsize) {
-        if (maxsize > 0) {
-            return getMaxSizeOfOversizedItems(mailboxAnnotations, maxsize);
+    private Optional<Integer> getMaxSizeValue(final List<MailboxAnnotation> mailboxAnnotations, Optional<Integer> maxsize) {
+        if (maxsize.isPresent()) {
+            return maxsize.transform(new Function<Integer, Optional<Integer>>() {
+                @Override
+                public Optional<Integer> apply(Integer input) {
+                    return getMaxSizeOfOversizedItems(mailboxAnnotations, input);
```

Comment:
Return type of getMaxSizeOfOversizedItems could be Integer. Like this you avoid unboxing, and avoid the if block.